components/ProductMatrix.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <div>
+    <div v-if="error">{{ error }}</div>
+    <div class="product-matrix">
+      <div v-for="card in cards" :key="card.sortKey">
+        <component :is="components[card.type]" :product="card" />
+      </div>
+    </div>
+    <v-row align="center">
+      <v-col class="text-center" cols="12">
+        <v-btn v-if="cards.length <= 6" :loading="loading" @click="loadMore">
+          Visa mer
+        </v-btn>
+      </v-col>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+/* eslint-disable dot-notation */
+import Vue from 'vue'
+import { mapGetters, mapActions } from 'vuex'
+import ProductCard from './ProductCard.vue'
+import InfoCard from './InfoCard.vue'
+import { CardTypes } from '~/types/Card'
+
+export default Vue.extend({
+  components: {
+    ProductCard
+  },
+  data() {
+    return {
+      error: '',
+      components: {
+        [CardTypes.PRODUCT]: ProductCard,
+        [CardTypes.INFO]: InfoCard
+      }
+    }
+  },
+  computed: mapGetters({
+    cards: 'cards/getCards',
+    loading: 'products/isLoading'
+  }),
+  mounted() {
+    this.init()
+  },
+  methods: mapActions({
+    init: 'products/init',
+    loadMore: 'products/loadMore'
```

Comment:
Isn't it better to have the loadMore function in the file where it's used?

Reply:
It's used on row 11

Reply:
Why not place the function (or what you call it) in `ProductMatrix.vue`?
````
    async loadMore({ dispatch, commit }) {
      const products = await dispatch('fetch', {
        placements: 'home_page.2020_start_full'
      })
      commit('gotProducts', products)
      commit('didShowMore')
    },
````